db/schema.rb

```diff
@@ -156,23 +156,39 @@
   add_index "map_positions", ["mapping_locus"], name: "map_positions_mapping_locus_idx", using: :btree
   add_index "map_positions", ["population_locus_id"], name: "map_positions_population_locus_id_idx", using: :btree
 
-  create_table "marker_assays", primary_key: "marker_assay_name", force: :cascade do |t|
-    t.text "canonical_marker_name", default: "unspecified", null: false
-    t.text "marker_type"
-    t.text "probe_name"
-    t.text "primer_a"
-    t.text "primer_b"
-    t.text "separation_system"
-    t.text "comments"
-    t.text "entered_by_whom"
-    t.date "date_entered"
-    t.text "data_provenance"
-    t.text "data_owned_by"
-    t.text "confirmed_by_whom"
-  end
-
-  create_table "marker_sequence_assignments", primary_key: "canonical_marker_name", force: :cascade do |t|
+  create_table "marker_assays", force: :cascade do |t|
+    t.text    "marker_assay_name",             default: "",            null: false
+    t.text    "canonical_marker_name",         default: "unspecified", null: false
+    t.text    "marker_type"
+    t.text    "primer_a_name"
+    t.text    "primer_b_name"
+    t.text    "separation_system"
+    t.text    "comments"
+    t.text    "entered_by_whom"
+    t.date    "date_entered"
+    t.text    "data_provenance"
+    t.text    "data_owned_by"
+    t.text    "confirmed_by_whom"
+    t.integer "restriction_enzyme_a_id"
+    t.integer "marker_sequence_assignment_id"
+    t.integer "restriction_enzyme_b_id"
+    t.integer "primer_a_id"
+    t.integer "primer_b_id"
+    t.integer "probe_id"
+  end
+
+  add_index "marker_assays", ["canonical_marker_name"], name: "marker_assays_canonical_marker_name_idx", using: :btree
+  add_index "marker_assays", ["marker_assay_name"], name: "marker_assays_marker_assay_name_idx", using: :btree
+  add_index "marker_assays", ["marker_sequence_assignment_id"], name: "marker_assays_marker_sequence_assignment_id_idx", using: :btree
+  add_index "marker_assays", ["primer_a_id"], name: "marker_assays_primer_a_id_idx", using: :btree
+  add_index "marker_assays", ["primer_a_name"], name: "marker_assays_primer_a_idx", using: :btree
+  add_index "marker_assays", ["primer_b_id"], name: "marker_assays_primer_b_id_idx", using: :btree
+  add_index "marker_assays", ["primer_b_name"], name: "marker_assays_primer_b_idx", using: :btree
+  add_index "marker_assays", ["probe_id"], name: "marker_assays_probe_id_idx", using: :btree
+
+  create_table "marker_sequence_assignments", force: :cascade do |t|
     t.text "marker_set",              default: "unspecified", null: false
+    t.text "canonical_marker_name",   default: "",            null: false
```

Comment:
It's the old PK of `marker_sequence_assignments`. `canonical_marker_name` was left as a text column in `marker_assays` for the usual reason (lots of unmatched records - removing this column would result in loss of data).